jupyterhub/handlers/base.py

```diff
@@ -1326,6 +1326,19 @@ async def stop_single_user(self, user, server_name=''):
         spawner = user.spawners[server_name]
         if spawner.pending:
             raise RuntimeError(f"{spawner._log_name} pending {spawner.pending}")
+
+        if self.authenticator.refresh_pre_stop:
+            auth_user = await self.refresh_auth(user, force=True)
+            if auth_user is None:
+                if self.current_user.name == user.name:
```

Comment:
needs to check that it's a user and not a service before checking the name, but 👍 

```suggestion
                if self.current_user.kind == "user" and self.current_user.name == user.name:
```